src/odig.ml

```diff
@@ -0,0 +1,59 @@
+open Import
+
+type t = Opam.t * Opam.Switch.t
+
+type of_opam_error = Odig_not_installed
+
+let of_opam (opam, switch) =
+  let open Promise.Syntax in
+  let+ is_installed = Opam.has_package opam switch "odig" in
+  if is_installed then
+    Ok (opam, switch)
+  else
+    Error Odig_not_installed
+
+let cmd_ouput (opam, switch) args =
+  let cmd = Opam.exec opam switch ~args in
+  Cmd.output cmd
+
+let cache_dir t =
+  let opam, switch = t in
+  let default_cache_dir = Path.(Opam.path opam switch / "/var/cache/odig/") in
```

Comment:
Where did you get the `/var/cache/odig` path from?

Reply:
From `odig conf --help`:

```
COMMON OPTIONS
       --cache-dir=PATH (absent ODIG_CACHE_DIR env)
           Cache directory. If unspecified, $PREFIX/var/cache/odig with
           $PREFIX the parent directory of odig's install directory.
```

My code might not be correct though if `odig` is not installed in the current opam switch 🤔 Should it be preferable to fail case we can't get the `cache_dir` from `odig`?

Reply:
I removed the `default_cache_dir`